src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py

```diff
@@ -738,27 +753,18 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
-        if output_type not in ["latent", "pt", "np", "pil"]:
-            deprecation_message = (
-                f"the output_type {output_type} is outdated. Please make sure to set it to one of these instead: "
-                "`pil`, `np`, `pt`, `latent`"
-            )
-            deprecate("Unsupported output_type", "1.0.0", deprecation_message, standard_warn=False)
-            output_type = "np"
-
-        if output_type == "latent":
-            image = latents
-            has_nsfw_concept = None
-
+        if not output_type == "latent":
+            image = self.vae.decode(latents / self.vae.config.scaling_factor).sample
+            image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
         else:
-            image = self.decode_latents(latents)
-
-            if self.safety_checker is not None:
-                image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
-            else:
-                has_nsfw_concept = False
-
-            image = self.image_processor.postprocess(image, output_type=output_type)
+            has_nsfw_concept = False
+        
+        do_normalize = (
+            [not has_nsfw for has_nsfw in has_nsfw_concept]
+            if isinstance(has_nsfw_concept, list)
+            else not has_nsfw_concept
+        )
+        image = self.image_processor.postprocess(image, output_type=output_type, do_normalize=do_normalize)
```

Comment:
@pcuenca could you also check here. Note we need this because of https://github.com/huggingface/diffusers/pull/2943#pullrequestreview-1390784152 IMO - the other idea is to detect whether an image is black in the postprocessor as discussed here: https://github.com/huggingface/diffusers/pull/2943#discussion_r1170677988 but IMO it's a bit too brittle

Reply:
Commented [above](https://github.com/huggingface/diffusers/pull/2943/files#r1174623438). I don't like to introduce black image detection in the post-processor so I'd go with this or by having `run_safety_checker` always return the same type of images (normalized).